pkg/ami/api.go

```diff
@@ -57,6 +57,20 @@ func IsAvailable(api ec2iface.EC2API, id string) (bool, error) {
 	return *output.Images[0].State == "available", nil
 }
 
+// LookupRootDeviceName gets the root block device mapping
+func LookupRootDeviceName(api ec2iface.EC2API, id string) (string, error) {
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []*string{&id},
+	}
+
+	output, err := api.DescribeImages(input)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to determine root device name for %q", id)
+	}
+
+	return *output.Images[0].RootDeviceName, nil
```

Comment:
We should check for lenght of the list, and return an error if it's not strictly one item. Also,  isn't it entirely possible for `RootDeviceName` to be unset, e.g. when the AMI doesn't actually have any particular requirements about the root volume?

Reply:
I agree that we should also check the length of the list. 

With regards to the RootDeviceName , EBS AMIs will always have RootDeviceName set but  instance-store AMIs will not. I will need to test but I think that the current launch config template will not work with instance-store AMIs if the VolumeSize is set in the config. I think that perhaps we need to check that the AMI is an EBS AMI(the vast majority of AMIs are EBS), what do you think?

Reply:
I have added the length check

Reply:
>  I think that perhaps we need to check that the AMI is an EBS AMI(the vast majority of AMIs are EBS), what do you think?

Yes, good idea!